app/src/main/java/fr/free/nrw/commons/CommonsApplication.kt

```diff
@@ -91,6 +92,13 @@ class CommonsApplication : MultiDexApplication() {
     @Inject
     lateinit var contributionDao: ContributionDao
 
+    /**
+     * Flag to check if speech recognition is available.
+     */
+    val isVoiceRecognitionAvailable by lazy {
+        SpeechRecognizer.isRecognitionAvailable(this)
```

Comment:
We're depending on this call to check if service is available or not. Since, I don't have access to the device, do you mind adding a print statement and see if it returns true, then likely issue is with API.